### PR TITLE
tests/netstat_l2: workaround for esp32 boards

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

Since ESP32 boards only have network interfaces if the `esp_wifi`, `esp_now` and `esp_eth` modules are explicitly enabled, they are removed from the list of boards that provide network interfaces. 

Instead, the variable BOARD_PROVIDES_NETIF is made adjustable in the environment. Thus the test can be used for ESP32 with and without network interfaces.

Fixes the test `tests/netstat_l2` for ESP32.

This PR belongs to the series of PRs, each with very small changes that fix automatic tests on ESP32 boards.

### Testing procedure

Make, flash and test with
```
BOARD_PROVIDES_NETIF=esp32-wroom-32 USEMODULE=esp_wifi make BOARD=esp32-wroom-32 -C tests/netstats_l2 flash test
```
and without network interfaces.
```
make BOARD=esp32-wroom-32 -C tests/netstats_l2
```

### Issues/PRs references

Requires #12752

### CODE REVIEW

**Comment by benpicco** [comment]
I was always wondering why you don't just do

```Makefile
ifneq (,$(filter gnrc_netdev_default netdev_default,$(USEMODULE)))
  USEMODULE += esp_wifi
endif
```

for esp32/esp8266 since WiFi is pretty much the defining feature of this MCU.

**Comment by gschorcht** [comment]
> I was always wondering why you don't just do
> 
> ```makefile
> ifneq (,$(filter gnrc_netdev_default netdev_default,$(USEMODULE)))
>   USEMODULE += esp_wifi
> endif
> ```
> 
> for esp32/esp8266 since WiFi is pretty much the defining feature of this MCU.

Because you have different options to use the WiFi interface, `esp_wifi` and `esp_now`. ESP32 also provides an Ethernet interface. Which one should be the default? So I decided to let the user define it.

**Comment by benpicco** [comment]
You could do `ESP_NETDEV_DEFAULT ?= esp_wifi` (or `esp_now`, afaik that doesn't need any additional config), then you could do

```Makefile
ifneq (,$(filter gnrc_netdev_default netdev_default,$(USEMODULE)))
  USEMODULE += $(ESP_NETDEV_DEFAULT )
endif
```

**Comment by miri64** [comment]
What to do if https://github.com/RIOT-OS/RIOT/pull/11676 is merged then?

File: tests/netstats_l2/Makefile [hunk 1]
```diff
@@ -1,10 +1,8 @@
 include ../Makefile.tests_common
 
-BOARD_PROVIDES_NETIF := airfy-beacon fox iotlab-m3 mulle native nrf51dk nrf51dongle \
+BOARD_PROVIDES_NETIF ?= airfy-beacon fox iotlab-m3 mulle native nrf51dk nrf51dongle \
```
Comment (aabadie) [inline tests/netstats_l2/Makefile]: Maybe use the following:
```suggestion
BOARD_PROVIDES_NETIF += airfy-beacon fox iotlab-m3 mulle native nrf51dk nrf51dongle \
```

Not sure if it's useful or not (if not you can ignore).

**Comment by aabadie** [comment]
> What to do if #11676 is merged then?

That won't happen in its current state. I think an adaption of @cladmi's approach in #12427 would be better (this is in my plans).

**Comment by gschorcht** [comment]
> You could do `ESP_NETDEV_DEFAULT ?= esp_wifi` (or `esp_now`, afaik that doesn't need any additional config), then you could do
> 
> ```makefile
> ifneq (,$(filter gnrc_netdev_default netdev_default,$(USEMODULE)))
>   USEMODULE += $(ESP_NETDEV_DEFAULT)
> endif
> ```

Should be possible in that way. But I wouldn't do that with this PR if you don't insist on this change because it changes the documented way on how to enable and select the network interface for ESPs.

File: tests/netstats_l2/Makefile (hunk 1 above)
Comment (gschorcht) [reply tests/netstats_l2/Makefile]: It should not make any difference if you specify this variable explicitly for its use with a single board. BTW, `netstat_l2` could be used with ESP8266 boards in same way.

**Comment by gschorcht** [comment]
@benpicco How do we continue with this PR?

**Comment by benpicco** [comment]
I mean if you were to make a PR to add a default netdev for esp*, this PR would be obsolete.
IMHO this would be the better solution, also from a user perspective.

**Comment by gschorcht** [comment]
Ok, I can try to find a usefull configuration. `esp_now` might be better as default since it doesn't require additional setting. But it might become a problem to activate `esp_wifi` which is probably used more often. I have to test.

**Comment by benpicco** [comment]
`esp_netdev_default` could be a pseudomodule that depends on `esp_now` if no other esp_wifi|esp_now|esp_ethernet module is used.

**Comment by gschorcht** [comment]
Hm, sounds good. I will try. BTW, `esp_wifi` and `esp_now` can also be used together, e.g., to realize a border router.

**Comment by benpicco** [comment]
> `esp_wifi` and `esp_now` can also be used together, e.g., to realize a border router.

Wow, I didn't know - this is neat!

**Comment by gschorcht** [comment]
Closed in favor of PR #12787.

**Pull request closed without merging**